stacks-signer/src/signerdb.rs

```diff
@@ -173,6 +177,17 @@ impl SignerDb {
         try_deserialize(result)
     }
 
+    /// Return the last signed block in a tenure (identified by its consensus hash)
+    pub fn get_last_signed_block_in_tenure(
+        &self,
+        tenure: &ConsensusHash,
+    ) -> Result<Option<BlockInfo>, DBError> {
+        let query = "SELECT block_info FROM blocks WHERE consensus_hash = ? AND signed_over = 1 ORDER BY stacks_height DESC LIMIT 1";
```

Comment:
You're going to want to index `block_info` on both `consensus_hash` and `signed_over`.

Reply:
Updated in fdcaea8b91972ad721d9d9e0579ac89fb531ddc0